src/main/java/com/atlassian/bitbucket/jenkins/internal/trigger/BitbucketWebhookMultibranchTrigger.java

```diff
@@ -0,0 +1,110 @@
+package com.atlassian.bitbucket.jenkins.internal.trigger;
+
+import com.atlassian.bitbucket.jenkins.internal.client.exception.BitbucketClientException;
+import com.atlassian.bitbucket.jenkins.internal.config.BitbucketPluginConfiguration;
+import com.atlassian.bitbucket.jenkins.internal.config.BitbucketServerConfiguration;
+import com.atlassian.bitbucket.jenkins.internal.model.BitbucketWebhook;
+import com.atlassian.bitbucket.jenkins.internal.scm.BitbucketSCMRepository;
+import com.atlassian.bitbucket.jenkins.internal.scm.BitbucketSCMSource;
+import hudson.Extension;
+import hudson.model.Item;
+import hudson.triggers.Trigger;
+import hudson.triggers.TriggerDescriptor;
+import jenkins.branch.MultiBranchProject;
+import org.jenkinsci.Symbol;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+import javax.inject.Inject;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import static java.util.Objects.requireNonNull;
+
+public class BitbucketWebhookMultibranchTrigger extends Trigger<MultiBranchProject<?, ?>> {
+
+    private static final Logger LOGGER = Logger.getLogger(BitbucketWebhookMultibranchTrigger.class.getName());
+
+    @SuppressWarnings("RedundantNoArgConstructor") // Required for Stapler
+    @DataBoundConstructor
+    public BitbucketWebhookMultibranchTrigger() {
+    }
+
+    @Override
+    public BitbucketWebhookMultibranchTrigger.BitbucketWebhookMultibranchTriggerDescriptor getDescriptor() {
+        return (BitbucketWebhookMultibranchTrigger.BitbucketWebhookMultibranchTriggerDescriptor) super.getDescriptor();
+    }
+
+    @Override
+    public void start(MultiBranchProject<?, ?> project, boolean newInstance) {
+        super.start(project, newInstance);
+        if (!newInstance) {
+            return;
+        }
+        BitbucketWebhookMultibranchTrigger.BitbucketWebhookMultibranchTriggerDescriptor descriptor = getDescriptor();
+        project.getSCMSources().stream().filter(scm -> scm instanceof BitbucketSCMSource)
+                .filter(scm -> ((BitbucketSCMSource) scm).isValid())
+                .forEach(scm -> {
+            boolean isAdded = descriptor.addTrigger(project, (BitbucketSCMSource) scm);
+            ((BitbucketSCMSource) scm).setWebhookRegistered(isAdded);
+        });
+    }
+
+    @Symbol("BitbucketWebhookMultibranchTriggerImpl")
+    @Extension
+    public static class BitbucketWebhookMultibranchTriggerDescriptor extends TriggerDescriptor {
```

Comment:
Could this simply be called `Descriptor`? Especially given we're always referencing it with the full name (including the parent class). This seems a bit unnecessarily verbose and redundant: `BitbucketWebhookMultibranchTrigger.BitbucketWebhookMultibranchTriggerDescriptor`

(we do something similar for builders that are inner classes, i.e. we just call them `Builder` and then reference them with the full type name)

Reply:
Did a quick scan through our code base and a few Jenkins classes, looks like `DescriptorImpl` is the standard, so I'll change it to that